sstables/sstables.cc

```diff
@@ -1954,15 +1954,15 @@ void sstable::validate_originating_host_id() const {
     }
 }
 
-future<> sstable::touch_temp_dir() {
-    if (_storage.temp_dir) {
+future<> sstable::filesystem_storage::touch_temp_dir(const sstable& sst) {
+    if (temp_dir) {
         return make_ready_future<>();
     }
-    auto temp_dir = get_temp_dir();
-    sstlog.debug("Touching temp_dir={}", temp_dir);
-    auto fut = sstable_touch_directory_io_check(temp_dir);
-    return fut.then([this, temp_dir = std::move(temp_dir)] () mutable {
-        _storage.temp_dir = std::move(temp_dir);
+    auto tmp = dir + "/" + sstable::sst_dir_basename(sst._generation);
```

Comment:
Sigh. We should really use filesytem::path for keeping and manipulating paths, not strings.
Can be done as a follow up of course

Reply:
Absolutely. But then the question is where to stop converting sstring to fs::path. I've a branch that makes fs::path appear on sstable and sstables_manager level and converts `sstring table::config::datadir` (containing full path) into `sstring table::config::location` cutting off the "/var/lib/scylla" part.

Reply:
makes sense, especially if the location is abstracted above the storage layer so it can be stored in the object ownership table.